zxlive/commands.py

```diff
@@ -142,6 +169,27 @@ def redo(self) -> None:
         self._added_vert = self.g.add_vertex(self.vty, self.y, self.x)
         self.update_graph_view()
 
+@dataclass
+class AddNonFlexNode(BaseCommand):
+    """Adds a new non-flexsymmetric node at a given position."""
+    x: float
+    y: float
+
+    _added_input_vert: Optional[VT] = field(default=None, init=False)
+    _added_output_vert: Optional[VT] = field(default=None, init=False)
+
+    def undo(self) -> None:
+        assert self._added_input_vert is not None
+        assert self._added_output_vert is not None
+        self.g.remove_vertex(self._added_input_vert)
+        self.g.remove_vertex(self._added_output_vert)
+        self.update_graph_view()
+
+    def redo(self) -> None:
+        self._added_input_vert = self.g.add_vertex(VertexType.W_INPUT, self.y-0.3, self.x)
```

Comment:
I don't like the hardcoded 0.3 here. Maybe make it a global constant or default member of the dataclass (i.e. declare `input_offset: float = 0.3`)?